### DISCUSSION TITLE
HistoricalLocation relation question

### DISCUSSION THREAD

**Discussion by charlycou** [opengeospatial/sensorthings#193] [Q&A] [answered]
HistoricalLocation entities have many-to-many relation with Location entities and many-to-one relation with Thing entities. 
I can't find the use case where a HistoricalLocation entity needs to be linked with several Location Entities of the same Thing entity. This would explain the relations mentioned before. In other words a Thing that can have several locations at the same time.
Can anyone clarify?

**Comment by hylkevds** [comment, marked as answer]
From https://docs.ogc.org/is/18-088/18-088.html#thing
Table 4:

> Location
> Many optional to many optional
> The Location entity locates the Thing. Multiple Things MAY be located at the same Location. A Thing MAY not have a Location. A Thing SHOULD have only one Location.
> 
> However, in some complex use cases, a Thing MAY have more than one Location representations. In such case, the Thing MAY have more than one Locations. These Locations SHALL have different encodingTypes and the encodingTypes SHOULD be in different spaces (e.g., one encodingType in Geometrical space and one encodingType in Topological space).

One example would be to have the Location as GeoJSON and WKT and text description.

Another example would be having several different resolutions for the geometries, though these have the same encodingTypes:
- Map: https://api4inspire.k8s.ilt-dmz.iosb.fraunhofer.de/servlet/is/163/
- France: https://demography.k8s.ilt-dmz.iosb.fraunhofer.de/v1.1/Things?$filter=properties/countryCode%20eq%20%27FR%27%20and%20properties/level%20eq%200&$expand=Locations

**Reply by charlycou** [reply]
Thank you @hylkevds. It is perfectly clear. Since you mention Table 4 from https://docs.ogc.org/is/18-088/18-088.html#thing, I am also wondering why it is specified that "a Thing SHOULD have only one Location"? The more complex use cases described after, do not suggest that a Thing could be moving.

**Reply by hylkevds** [reply]
A thing has only one real-world location, since things can't be in two places at the same time. But there may be multiple ways to describe the place the thing is in, hence the multiple Location entities.

The multiple Location entities have nothing to do with the thing moving. That's what HistoricalLocation is for. It links the Thing to the places it has been. And since a place can have multiple ways to describe it, the HistoricalLocation is linked to multiple Locations too, just like the Thing.

**Reply by charlycou** [reply]
Understood thank you for the precision. It make sense now. A Thing should be attached to only one Location unless this Location is described in different ways. Ancient Location of the Thing must not be associated to the Thing anymore but only to HistoricalLocation entity itself linked to the Thing. I missed this last part.
